fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStore.java

```diff
@@ -3039,12 +3049,23 @@ private RecordMetaDataProvider getMetaDataProviderForBuild() {
 
         @Nonnull
         private CompletableFuture<Void> preloadMetaData() {
-            if (metaDataStore != null) {
-                return metaDataStore.preloadMetaData(metaDataProvider);
+            if (metaDataStore != null && localMetaDataFileDescriptor != null) {
+                return metaDataStore.preloadMetaData(metaDataProvider, localMetaDataFileDescriptor);
+            } else if (metaDataStore != null) {
+                return metaDataStore.preloadMetaData(metaDataProvider, metaDataProvider != null ?
+                                                                       metaDataProvider.getRecordMetaData().getRecordsDescriptor() : null);
             } else {
                 return AsyncUtil.DONE;
             }
         }
 
+        @Nullable
+        public Descriptors.FileDescriptor getLocalMetaDataFileDescriptor() {
+            return localMetaDataFileDescriptor;
+        }
+
+        public void setLocalMetaDataFileDescriptor(@Nullable Descriptors.FileDescriptor localMetaDataFileDescriptor) {
+            this.localMetaDataFileDescriptor = localMetaDataFileDescriptor;
```

Comment:
The getter and setter for `localMetaDataFileDescriptor` should be moved to live next to the other getters and setters; the setter should return `this` to match the builder pattern; and then I believe the `BaseBuilder` implementations need versions of the setter in order to override the return type (until Java implements [`SELF_TYPE`](https://dijkstra.eecs.umich.edu/eecs483/cool-manual/node8.html)).

Reply:
oh good catch! Thanks!